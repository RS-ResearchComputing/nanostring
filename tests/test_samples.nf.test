nextflow_pipeline {

    name "Test Workflow main.nf - NANOSTRING (test_samples)"
    script "../main.nf"
    profile "+test_samples"
    tag "test_samples"
    tag "pipeline"

    test("test_samples profile") {

        when {
            params {
                outdir        = "$outputDir"
            }
        }

        then {
            // stable_name: All files + folders in ${params.outdir}/ with a stable name
            // Ignore files with timestamps in their names or with unstable prefix No_HK_
            def stable_name = getAllFilesFromDir(params.outdir, relative: true, includeDir: true, ignore: [
                'pipeline_info/*.{html,json,txt}',
                'annotated_tables/*_normalized_counts_Norm_GEX_ENDO.tsv',
                'annotated_tables/*_normalized_counts_Norm_GEX_HK.tsv',
                'normalized_counts/*_normalized_counts.tsv',
                'normalized_counts/*_normalized_counts_wo_HKnorm.tsv'
                ])
            // stable_path: All files in ${params.outdir}/ with stable content
            def stable_path = getAllFilesFromDir(params.outdir, ignoreFile: 'tests/.nftignore')
            assertAll(
                { assert workflow.success},
                { assert snapshot(
                    // Number of successful tasks
                    workflow.trace.succeeded().size(),
                    // pipeline versions.yml file for multiqc from which Nextflow version is removed because we tests pipelines on multiple Nextflow versions
                    removeNextflowVersion("$outputDir/pipeline_info/nf_core_pipeline_software_mqc_versions.yml"),
                    // All stable path name, with a relative path
                    stable_name,
                    // All files with stable contents
                    stable_path
                ).match() }
            )
        }
    }
}
