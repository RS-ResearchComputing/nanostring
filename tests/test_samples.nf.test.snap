{
    "test_samples profile": {
        "content": [
            7,
            {
                "CREATE_ANNOTATED_TABLES": {
                    "r-base": "4.3.3",
                    "r-ggplot2": null,
                    "r-dplyr": "1.1.4",
                    "r-readr": "2.1.5"
                },
                "CREATE_GENE_HEATMAP": {
                    "r-base": "4.2.3",
                    "r-dplyr": "1.1.4",
                    "r-ggplot2": "3.4.4",
                    "r-rlang": "1.1.1",
                    "bioconductor-ComplexHeatmap": "2.14.0",
                    "r-circlize": "0.4.15",
                    "r-yaml": "2.3.8",
                    "r-fs": "1.6.4"
                },
                "NACHO_NORMALIZE": {
                    "r-base": "4.3.3",
                    "r-nacho": "2.0.6",
                    "r-dplyr": "1.1.4",
                    "r-ggplot2": "3.4.4",
                    "r-tidyr": "1.3.0",
                    "r-readr": "2.1.5",
                    "r-fs": "1.6.4",
                    "r-optparse": "1.7.5"
                },
                "NACHO_QC": {
                    "r-base": "4.3.3",
                    "r-nacho": "2.0.6",
                    "r-dplyr": "1.1.4",
                    "r-ggplot2": "3.4.4",
                    "r-tidyr": "1.3.0",
                    "r-readr": "2.1.5",
                    "r-fs": "1.6.4",
                    "r-optparse": "1.7.5"
                },
                "Workflow": {
                    "nf-core/nanostring": "v1.3.1dev"
                }
            },
            [
                "QC",
                "QC/NACHO",
                "QC/NACHO/NanoQC.html",
                "QC/NACHO/NanoQC_with_outliers.html",
                "QC/NACHO/hk_detected_mqc.txt",
                "QC/NACHO/normalized_qc_mqc.txt",
                "QC/NACHO/png",
                "QC/NACHO/png/AVG_vs_BD_mqc.png",
                "QC/NACHO/png/AVG_vs_MED_mqc.png",
                "QC/NACHO/png/BD_mqc.png",
                "QC/NACHO/png/FOV_mqc.png",
                "QC/NACHO/png/HKF_mqc.png",
                "QC/NACHO/png/HK_mqc.png",
                "QC/NACHO/png/LOD_mqc.png",
                "QC/NACHO/png/Neg_mqc.png",
                "QC/NACHO/png/PCA1_vs_PCA2_mqc.png",
                "QC/NACHO/png/PCA_mqc.png",
                "QC/NACHO/png/PCAi_mqc.png",
                "QC/NACHO/png/POSF_vs_NEGF_mqc.png",
                "QC/NACHO/png/Pos_mqc.png",
                "QC/NACHO/png/Pos_vs_neg_mqc.png",
                "QC/NACHO/png/Posctrl_linearity_mqc.png",
                "QC/NACHO/png/plot_normf_mqc.png",
                "annotated_tables",
                "annotated_tables/counts_Norm_GEX_ENDO.tsv",
                "annotated_tables/counts_Norm_GEX_HK.tsv",
                "annotated_tables/counts_wo_HKnorm_Norm_GEX_ENDO.tsv",
                "annotated_tables/counts_wo_HKnorm_Norm_GEX_HK.tsv",
                "gene_heatmaps",
                "gene_heatmaps/gene_heatmap_mqc.png",
                "gene_heatmaps/wo_HKnorm_gene_heatmap_mqc.png",
                "multiqc",
                "multiqc/multiqc_data",
                "multiqc/multiqc_data/multiqc.log",
                "multiqc/multiqc_data/multiqc_No_HK_Norm_GEX_ENDO.txt",
                "multiqc/multiqc_data/multiqc_No_HK_Norm_GEX_HK.txt",
                "multiqc/multiqc_data/multiqc_Norm_GEX_ENDO.txt",
                "multiqc/multiqc_data/multiqc_Norm_GEX_HK.txt",
                "multiqc/multiqc_data/multiqc_citations.txt",
                "multiqc/multiqc_data/multiqc_data.json",
                "multiqc/multiqc_data/multiqc_normalized_qc.txt",
                "multiqc/multiqc_data/multiqc_software_versions.txt",
                "multiqc/multiqc_data/multiqc_sources.txt",
                "multiqc/multiqc_plots",
                "multiqc/multiqc_plots/pdf",
                "multiqc/multiqc_plots/pdf/No_HK_Norm_GEX_ENDO.pdf",
                "multiqc/multiqc_plots/pdf/No_HK_Norm_GEX_HK.pdf",
                "multiqc/multiqc_plots/pdf/Norm_GEX_ENDO.pdf",
                "multiqc/multiqc_plots/pdf/Norm_GEX_HK.pdf",
                "multiqc/multiqc_plots/pdf/normalized_qc.pdf",
                "multiqc/multiqc_plots/png",
                "multiqc/multiqc_plots/png/No_HK_Norm_GEX_ENDO.png",
                "multiqc/multiqc_plots/png/No_HK_Norm_GEX_HK.png",
                "multiqc/multiqc_plots/png/Norm_GEX_ENDO.png",
                "multiqc/multiqc_plots/png/Norm_GEX_HK.png",
                "multiqc/multiqc_plots/png/normalized_qc.png",
                "multiqc/multiqc_plots/svg",
                "multiqc/multiqc_plots/svg/No_HK_Norm_GEX_ENDO.svg",
                "multiqc/multiqc_plots/svg/No_HK_Norm_GEX_HK.svg",
                "multiqc/multiqc_plots/svg/Norm_GEX_ENDO.svg",
                "multiqc/multiqc_plots/svg/Norm_GEX_HK.svg",
                "multiqc/multiqc_plots/svg/normalized_qc.svg",
                "multiqc/multiqc_report.html",
                "normalized_counts",
                "normalized_counts/normalized_counts.tsv",
                "normalized_counts/normalized_counts_wo_HKnorm.tsv",
                "pipeline_info",
                "pipeline_info/nf_core_nanostring_software_mqc_versions.yml"
            ],
            [
                "hk_detected_mqc.txt:md5,61209383acc2abf6fc3ea309b5a5e094",
                "normalized_qc_mqc.txt:md5,9a0b015a28094a17331b12b08898da8e",
                "AVG_vs_BD_mqc.png:md5,06ba42cbbbbbfebd04cdcc1e609f5945",
                "AVG_vs_MED_mqc.png:md5,783d2149574bd725e595e1fadeb84927",
                "HKF_mqc.png:md5,a21a91e53904e74a81da92eff0260816",
                "PCA1_vs_PCA2_mqc.png:md5,a07d944ed0ed85f814143eebcf8ae507",
                "PCA_mqc.png:md5,8e4456b730ab72f0dcd5ed1895660df8",
                "PCAi_mqc.png:md5,2338d5fe0e447fc74d4969725546c136",
                "POSF_vs_NEGF_mqc.png:md5,0e1b5105e68a6e0c51471afce218abdb",
                "plot_normf_mqc.png:md5,6ffd3c11f8f5ddfbd3715e1780b10b01",
                "multiqc_No_HK_Norm_GEX_ENDO.txt:md5,ebd78401fb548c820ee7e678d2d2aec4",
                "multiqc_No_HK_Norm_GEX_HK.txt:md5,0e65e55682d5a7e3879f62de630ab2aa",
                "multiqc_Norm_GEX_ENDO.txt:md5,11388cd9088a4f902a6b8fe499f0c379",
                "multiqc_Norm_GEX_HK.txt:md5,784a91d6ada77b737f52233734b67afb",
                "multiqc_citations.txt:md5,4c806e63a283ec1b7e78cdae3a923d4f",
                "multiqc_normalized_qc.txt:md5,44a0350c527fe8735fc6b419536a78f5",
                "multiqc_sources.txt:md5,d2a044df39ce3c6abe5cdc2d67473490"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-01-20T14:04:00.375607"
    }
}